cabot/metricsapp/models/elastic.py

```diff
@@ -37,3 +49,132 @@ def client(self):
         if self._client:
             return self._client
         return create_es_client(self.urls, self.timeout)
+
+
+class ElasticsearchStatusCheck(MetricsStatusCheckBase):
+    class Meta:
+        app_label = 'metricsapp'
+
+    queries = models.TextField(
+        max_length=10000,
+        help_text='List of raw json Elasticsearch queries. Format: [q] or [q1, q2, ...]. '
+                  'Query guidelines: all aggregations should be named "agg." The most '
+                  'internal aggregation must be a date_histogram. Metrics names should be '
+                  'the same as the metric types (e.g., "max", "min", "avg").'
+    )
+
+    def get_series(self):
+        """
+        Get the relevant data for a check from Elasticsearch and parse it
+        into a generic format.
+        :param check: the ElasticsearchStatusCheck
+        :return data in the format
+            status:
+            error_message:
+            error_code:
+            raw:
+            data:
+              - series: a.b.c.d
+                datapoints:
+                  - [timestamp, value]
+                  - [timestamp, value]
+              - series: a.b.c.p.q
+                datapoints:
+                  - [timestamp, value]
+                check:
+        """
+        parsed_data = dict()
+        parsed_data['raw'] = None
+        # Will be set to true if we encounter an error
+        parsed_data['error'] = False
+        parsed_data['data'] = []
+
+        try:
+            queries = json.loads(self.queries)
+        except ValueError as e:
+            logger.exception('Error loading Elasticsearch queries: {}'.format(self.queries))
+            parsed_data['error_code'] = type(e).__name__
+            parsed_data['error_message'] = str(e)
+            parsed_data['error'] = True
+            return parsed_data
+
+        for query in queries:
+            source = ElasticsearchSource.objects.get(name=self.source.name)
+            try:
+                search = Search().from_dict(query)
+                response = search.using(source.client).index(source.index).execute()
+                parsed_data['raw'] = response.to_dict()
```

Comment:
response.to_dict() is called twice. We need to avoid it.

```python
raw_data = response.to_dict()
parsed_data['raw'] = raw_data
parsed_data['data'].extend(self._es_rec(raw_data['aggregations']))
```